Gem/Settings/ViewModels/ChainSettingsViewModel.swift

```diff
@@ -4,54 +4,149 @@ import Foundation
         self.selectedExplorer = explorerService.get(chain: chain) ?? explorers.first
     }
-    
-    var title: String {
-        Asset(chain).name
+
+    var title: String { Asset(chain).name }
+
+    var nodesTitle: String { Localized.Settings.Networks.source }
+    var nodesModels: [ChainNodeViewModel] {
+        nodes.map { node in
+            let chainData = nodesChainIdData[node.id]
+            return ChainNodeViewModel(
+                chainNode: node,
+                blockNumber: chainData?.blockNumber ?? .none,
+                latency: chainData?.latency ?? .none,
+                blockNumberError: chainData?.error
+            )
+        }
     }
 
+    var explorerTitle: String { Localized.Settings.Networks.explorer }
+
     var isSupportedAddingCustomNode: Bool {
         AssetConfiguration.addCustomNodeChains.contains(chain.type)
     }
+
+    func canDelete(chainNode: ChainNode) -> Bool {
+        guard nodes.count > 1 else { return false }
+        return !chainNode.isGemNode
+    }
 }
 
 // MARK: - Business logic
 
 extension ChainSettingsViewModel {
-    func getNodes() throws -> [ChainNode] {
-        try nodeService.nodes(for: chain)
+    func fetch() async throws {
+        try await MainActor.run { [self] in
+            nodes = try nodeService.nodes(for: chain)
+        }
+        await fetchAvailableChainIds()
+    }
+
+    func selectNode(node: ChainNode) throws {
+        selectedNode = node
+        try nodeService.setNodeSelected(chain: chain, node: selectedNode.node)
+    }
+
+    func deleteNode() throws {
+        guard let nodeDelete else { return }
+        let shouldReselectNode = nodeDelete == selectedNode
+        try nodeService.delete(node: nodeDelete.node, chain: chain)
+        if shouldReselectNode {
+            if let nodeToSelect = nodes.first(where: { $0.isGemNode }) ?? nodes.first(where: { $0 != nodeDelete }) {
+                try nodeService.setNodeSelected(chain: chain, node: nodeToSelect.node)
+                selectedNode = nodeService.getNodeSelected(chain: chain)
+            }
+        }
+        nodes = try nodeService.nodes(for: chain)
     }
 
     func selectExplorer(name: String) {
         selectedExplorer = name
         explorerService.set(chain: chain, name: name)
     }
 }
+
+// MARK: - Private
+
+extension ChainSettingsViewModel {
+    private func fetchAvailableChainIds() async {
+        guard !nodes.isEmpty else { return }
+        await withTaskGroup(of: (ChainFetchData).self) { [self] group in
+            for node in nodes {
+                group.addTask { [self] in
+                    await fetchChainId(for: node)
+                }
+            }
+
+            for await (chainFetchData) in group {
+                await MainActor.run { [self] in
+                    self.nodesChainIdData[chainFetchData.chainNode.id] = chainFetchData
+                }
+            }
+        }
+    }
+
+    private func fetchChainId(for node: ChainNode) async -> (ChainFetchData) {
```

Comment:
can you create struct for `ChainFetchData`, typealias creates for complexity here to deal with